### ISSUE TITLE
sparse vector divide by float gives full vector

### ISSUE THREAD

**Issue by augustinyiptong** [JuliaLang/julia#14111]
Hi

I have:
```jl
julia> s = sparsevec([1.0, 2.0, 3.0])
Sparse vector of length 3, with 3 Float64 nonzero entries:
  [1]  =  1.0
  [2]  =  2.0
  [3]  =  3.0
```

I do:
```jl
julia> s * 0.5
Sparse vector of length 3, with 3 Float64 nonzero entries:
  [1]  =  0.5
  [2]  =  1.0
  [3]  =  1.5
```

So far so good. If i do:
```jl
julia> s/2.0
 3-element Array{Float64,1}:
 0.5
 1.0
 1.5
```

Why does it get converted to a full vector?

I am using Julia Version `0.5.0-dev+1158`

Thanks, Gus

**Comment by nalimilan** [comment]
Looks like an oversight, as sparsity is preserved for sparse matrices.

@augustinyiptong If you call `@which s.*1.0` and `@less s.*1.0`, you'll see that multiplication is defined very simply by calling `scale`. This can be a good workaround at least, multiplying by `1/a`.

A better fix would be to add a function like this:
```julia
import Base.SparseArrays.nonzeroinds
./(x::AbstractSparseVector, a::Number) = SparseVector(length(x), copy(nonzeroinds(x)), nonzeros(x)./a)
```

**Comment by tkelman** [comment]
be sure to check for a zero denominator.

**Comment by ViralBShah** [comment]
We can do the obvious fix for now - but I think sparse solvers can generally do better with sparse RHS. I'm not sure what the state of the art is on this question today though.

**Comment by tkelman** [comment]
Unless you matrix is diagonal, solving a linear system of equations doesn't generally preserve sparsity in the right hand side. Simplex is the usual poster child algorithm where sparse vectors play a crucial role.

This issue is just about division by a scalar though, so "solvers" aren't so relevant here.

**Comment by tkelman** [comment]
closed by #15579
